File: scm-csharp/scheme/SchemeServer.cs
using System.Net.Sockets;
using System.Threading;
using System.Threading.Tasks;

namespace scheme;

public class SchemeServer
{
    public CancellationTokenSource cts;
    public Task task;
    public TcpListener listener;
    public SemaphoreSlim? sem;
    public int maxThreads;
    public int gracefulStopMs;

    public SchemeServer(CancellationTokenSource cts, Task task, TcpListener listener)
    {
        this.cts = cts;
        this.task = task;
        this.listener = listener;
    }

    public override string ToString() => "#<server>";
}

File: scm-java/src/main/scheme/SchemeServer.java
package scheme;

import java.net.ServerSocket;
import java.util.concurrent.Semaphore;
import java.util.concurrent.atomic.AtomicBoolean;
import java.util.concurrent.ExecutorService;

public class SchemeServer {
    public AtomicBoolean running = new AtomicBoolean(true);
    public Thread thread;
    public ExecutorService executor;
    public ServerSocket serverSocket;
    public Semaphore sem;
    public int maxThreads;
    public int gracefulStopMs;

    public SchemeServer(Thread t) {
        this.thread = t;
    }

    @Override
    public String toString() { return "#<server>"; }
}
